File: sdk/nodejs/cmd/pulumi-language-nodejs/testdata/published/bun/projects/l3-component-nested/innerComponent.ts
import * as pulumi from "@pulumi/pulumi";
import * as simple from "@pulumi/simple";

interface InnerComponentArgs {
    /**
     * An input passed to the inner component
     */
    input: pulumi.Input<boolean>,
}

export class InnerComponent extends pulumi.ComponentResource {
    public output: pulumi.Output<boolean>;
    constructor(name: string, args: InnerComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        super("components:index:InnerComponent", name, args, opts);
        const res = new simple.Resource(`${name}-res`, {value: !args.input}, {
            parent: this,
        });

        this.output = res.value;
        this.registerOutputs({
            output: res.value,
        });
    }
}

File: sdk/python/cmd/pulumi-language-python/testdata/toml/local/projects/l3-component-nested/innerComponent.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import builtins as _builtins
import pulumi_simple as simple

class InnerComponentArgs(TypedDict):
    input: Input[_builtins.bool]

class InnerComponent(pulumi.ComponentResource):
    def __init__(self, name: str, args: InnerComponentArgs, opts:Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:InnerComponent", name, args, opts)

        res = simple.Resource(f"{name}-res", value=not args["input"],
        opts = pulumi.ResourceOptions(parent=self))

        self.output = res.value
        self.register_outputs({
            'output': res.value
        })